docs/contributor/image_mode.md

```diff
@@ -0,0 +1,352 @@
+terminal session.
+
+The host shares the following configuration with the container:
+* The `openvswitch` kernel module to be used by the Open vSwitch service
+* A pull secret file for downloading the required OpenShift container images
+* Host container storage for reusing available container images
+
+```
+PULL_SECRET=~/.pull-secret.json
+IMAGE_NAME=microshift-4.16-bootc
+
+sudo modprobe openvswitch
+sudo podman run --rm -it --privileged \
+    -v "${PULL_SECRET}":/etc/crio/openshift-pull-secret:ro \
+    -v /var/lib/containers/storage:/var/lib/containers/storage \
+    --name "${IMAGE_NAME}" \
+    "${IMAGE_NAME}"
+```
+
+> The `systemd-modules-load` service will fail to start in the container if the
+> host kernel version is different from the `bootc` image kernel version. This
+> failure can be safely ignored as all the necessary kernel modules have already
+> been loaded by the host.
+
+> If additional LVM volume group device was allocated as described in the
+> [Configure CSI](#configure-csi) section, the loop device should automatically
+> be shared with the container and used by the MicroShift CSI driver.
+
+After the MicroShift `bootc` image has been successfully started, a login prompt
+will be presented in the terminal. Log into the running container using the
+`redhat:<password>` credentials.
+
+Run the following command to verify that all the MicroShift pods are up and running
+without errors.
+
+```
+watch sudo oc get pods -A \
+    --kubeconfig /var/lib/microshift/resources/kubeadmin/kubeconfig
+```
+
+> Run the `sudo shutdown now` command to stop the container.
+
+## Run MicroShift Bootc Virtual Machine
+
+Log into the `physical hypervisor host` using the user credentials that have
+SUDO permissions configured.
+
+### Prepare Kickstart File
+
+Run the following commands to create the `kickstart.ks` file to be used during
+the virtual machine installation.
+
+> The pull secret file contents are embedded in `kickstart.ks` and they are used
+> to create the `/etc/crio/openshift-pull-secret` file at the post-install stage.
+
+```
+PULL_SECRET=~/.pull-secret.json
+
+cat > kickstart.ks <<EOFKS
+lang en_US.UTF-8
+keyboard us
+timezone UTC
+text
+reboot
+
+# Partition disk with a 1GB boot XFS partition and an LVM volume containing a 10GB+
+# system root. The remainder of the volume is for the CSI driver for storing data.
+zerombr
+clearpart --all --initlabel
+part biosboot --fstype=biosboot --size=1 --asprimary
+part /boot/efi --fstype=efi --size=200
+part /boot --fstype=xfs --asprimary --size=800
+part pv.01 --grow
+volgroup rhel pv.01
+logvol / --vgname=rhel --fstype=xfs --size=10240 --name=root
+
+# Lock root user account
+rootpw --lock
+
+# Configure network to use DHCP and activate on boot
+network --bootproto=dhcp --device=link --activate --onboot=on
+
+# Pull a 'bootc' image from a remote registry
+ostreecontainer --url quay.io/myorg/mypath/microshift-4.16-bootc
```

Comment:
How does this authenticate? Does image needs to be public? Isn't this leaking subscription protected assets?

Reply:
Good catch. I'll make some changes.